rp2040-hal/src/vector_table.rs

```diff
@@ -0,0 +1,84 @@
+//! Interrupt vector table utilities
+//!
+//! Provide functionality to switch to another vector table using the
+//! Vector Table Offset Register (VTOR) of the Cortex-M0+
+//! Also provides types and utilities for copying a vector table into RAM
+
+/// Entry for a Vector in the Interrupt Vector Table.
+///
+/// Each entry in the Vector table is a union with usize to allow it to be 0 initialized via const initializer
+///
+/// Implementation borrowed from https://docs.rs/cortex-m-rt/0.7.1/cortex_m_rt/index.html#__interrupts
+#[derive(Clone, Copy)]
+pub union Vector {
```

Comment:
Does this need to be `pub`?

Reply:
No, this was me being lazy when copying the code.
Will remove pub, user doesn't need to see this.